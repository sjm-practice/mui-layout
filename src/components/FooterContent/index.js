import React from "react";
import Divider from "@material-ui/core/Divider";
import Grid from "@material-ui/core/Grid";
import MuiLink from "@material-ui/core/Link";
import Typography from "@material-ui/core/Typography";

function Copyright() {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {"Copyright Â© "}
      <MuiLink color="inherit" href="https://www.google.com">
        COMPANY_NAME
      </MuiLink>
      {` ${new Date().getFullYear()} .`}
    </Typography>
  );
}

const FooterContent = () => (
  <div style={{ maxWidth: 700, margin: "auto", textAlign: "center" }}>
    <Copyright />
    <Divider style={{ margin: "24px auto", width: 60 }} />
    <Grid container justify="center" spacing={2}>
      <Grid item xs={12} sm={6} md={3}>
        <Typography align="center" gutterBottom color="textSecondary">
          About
        </Typography>
      </Grid>
      <Grid item xs={12} sm={6} md={3}>
        <Typography align="center" gutterBottom color="textSecondary">
          Terms & Conditions
        </Typography>
      </Grid>
      <Grid item xs={12} sm={6} md={3}>
        <Typography align="center" gutterBottom color="textSecondary">
          Contact us
        </Typography>
      </Grid>
    </Grid>
  </div>
);

export default FooterContent;
