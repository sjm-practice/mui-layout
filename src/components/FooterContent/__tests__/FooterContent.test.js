/* eslint-env jest */

import React from "react";
import { render } from "@testing-library/react";

import FooterContent from "../index";

describe("With React Testing Library", () => {
  it.skip('Should show "Welcome!!"', () => {
    const { getByText } = render(<FooterContent />);

    expect(getByText("Welcome!!")).not.toBeNull();
  });
});

describe("With React Testing Library Snapshot", () => {
  it.skip("Should match Snapshot", () => {
    const { asFragment } = render(<FooterContent />);

    expect(asFragment()).toMatchSnapshot();
  });
});
