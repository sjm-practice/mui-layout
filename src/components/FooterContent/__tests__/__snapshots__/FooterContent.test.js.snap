// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterContent /> Should match Snapshot 1`] = `
<DocumentFragment>
  <div
    style="max-width: 700px; margin: auto; text-align: center;"
  >
    <span
      class="MuiTypography-root MuiTypography-caption MuiTypography-alignCenter"
    >
      Â© Copyright 2019
    </span>
    <hr
      class="MuiDivider-root"
      style="margin: 24px auto; width: 60px;"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom MuiTypography-alignCenter"
        >
          About
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom MuiTypography-alignCenter"
        >
          Blog
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom MuiTypography-alignCenter"
        >
          Terms & Conditions
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom MuiTypography-alignCenter"
        >
          Contact us
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
